### PR TITLE
Add javadocs for symbolic execution classes

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: javascript-frontend/src/main/java/org/sonar/javascript/se/ExpressionStack.java [hunk 1]
```diff
@@ -39,7 +39,11 @@
 import org.sonar.plugins.javascript.api.tree.expression.LiteralTree;
 import org.sonar.plugins.javascript.api.tree.expression.TemplateLiteralTree;
 
-class ExpressionStack {
+/**
+ * This class stores the stack of symbolic values corresponding to the order of expression evaluation.
+ * Each {@link ProgramState} has corresponding instance of {@link ExpressionStack}.
+ */
```
Comment (pynicolas) [inline javascript-frontend/src/main/java/org/sonar/javascript/se/ExpressionStack.java]: @vilchik-elena This class is immutable.

File: javascript-frontend/src/main/java/org/sonar/javascript/se/ExpressionStack.java [hunk 2]
```diff
@@ -67,6 +71,12 @@ public ExpressionStack push(@Nullable SymbolicValue newValue) {
     return new ExpressionStack(newStack);
   }
 
+  /**
+   * This method executes expression: it pushes to the stack a new symbolic value based (if required) on popped symbolic values.
```
Comment (pynicolas) [inline javascript-frontend/src/main/java/org/sonar/javascript/se/ExpressionStack.java]: @vilchik-elena It should be clearer that the current object is *not* modified.

File: javascript-frontend/src/main/java/org/sonar/javascript/se/ProgramState.java [hunk 3]
```diff
@@ -37,6 +37,9 @@
 import org.sonar.plugins.javascript.api.symbols.Symbol;
 import org.sonar.plugins.javascript.api.tree.expression.ExpressionTree;
 
+/**
+ * This class represents the knowledge about the variables values at some program point.
```
Comment (pynicolas) [inline javascript-frontend/src/main/java/org/sonar/javascript/se/ProgramState.java]: @vilchik-elena ProgramState does not have any knowledge of a program point and the same program state may be valid for several program points.

File: javascript-frontend/src/main/java/org/sonar/javascript/se/ExpressionStack.java (hunk 2 above)
Comment (vilchik-elena) [inline javascript-frontend/src/main/java/org/sonar/javascript/se/ExpressionStack.java]: @pynicolas do you have suggestion how to rewrite this part then?

File: javascript-frontend/src/main/java/org/sonar/javascript/se/ExpressionStack.java (hunk 1 above)
Comment (vilchik-elena) [inline javascript-frontend/src/main/java/org/sonar/javascript/se/ExpressionStack.java]: @pynicolas fixed a78b43ec61b612338b672bfd482ee56721ebfaf5

File: javascript-frontend/src/main/java/org/sonar/javascript/se/ProgramState.java (hunk 3 above)
Comment (vilchik-elena) [inline javascript-frontend/src/main/java/org/sonar/javascript/se/ProgramState.java]: @pynicolas fixed a78b43ec61b612338b672bfd482ee56721ebfaf5

**Pull request merged**